File: backend/src/main/java/com/paysphere/entity/LinkedAccount.java
package com.paysphere.entity;

import com.paysphere.enums.LinkedAccountStatus;
import com.paysphere.enums.LinkedAccountType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "linked_accounts")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class LinkedAccount extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(name = "account_name", nullable = false, length = 100)
    private String accountName;

    @Column(name = "bank_name", nullable = false, length = 100)
    private String bankName;

    @Column(name = "account_number", nullable = false, length = 30)
    private String accountNumber;

    @Column(name = "routing_number", length = 20)
    private String routingNumber;

    @Enumerated(EnumType.STRING)
    @Column(name = "account_type", nullable = false, length = 20)
    @Builder.Default
    private LinkedAccountType accountType = LinkedAccountType.CHECKING;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false, length = 20)
    @Builder.Default
    private LinkedAccountStatus status = LinkedAccountStatus.PENDING;

    @Column(name = "is_primary", nullable = false)
    @Builder.Default
    private Boolean isPrimary = false;
}

File: frontend/src/types/linkedAccount.ts
export type LinkedAccountType = 'CHECKING' | 'SAVINGS' | 'BUSINESS';
export type LinkedAccountStatus = 'PENDING' | 'VERIFIED' | 'FAILED' | 'REMOVED';

export interface LinkedAccount {
  id: string;
  accountName: string;
  bankName: string;
  maskedAccountNumber: string;
  routingNumber?: string;
  accountType: LinkedAccountType;
  status: LinkedAccountStatus;
  primary: boolean;
  createdAt: string;
}

export interface LinkedAccountRequest {
  accountName: string;
  bankName: string;
  accountNumber: string;
  routingNumber?: string;
  accountType?: string;
  isPrimary?: boolean;
}

export interface UpdateProfileRequest {
  firstName?: string;
  lastName?: string;
  phone?: string;
  avatarUrl?: string;
}
